src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiResponseGenerator.cs

```diff
@@ -72,35 +73,79 @@ public static OpenApiResponse GetResponse(this string statusCode)
+        public static OpenApiResponses CreateResponses(this ODataContext context, IEdmOperationImport operationImport, ODataPath path)
         {
             Utils.CheckArgumentNull(context, nameof(context));
             Utils.CheckArgumentNull(operationImport, nameof(operationImport));
+            Utils.CheckArgumentNull(path, nameof(path));
 
-            return context.CreateResponses(operationImport.Operation);
+            return context.CreateResponses(operationImport.Operation, path);
         }
 
         /// <summary>
         /// Create the <see cref="OpenApiResponses"/> for a <see cref="IEdmOperation"/>
         /// </summary>
         /// <param name="context">The OData context.</param>
         /// <param name="operation">The Edm operation.</param>
+        /// <param name="path">The OData path.</param>
         /// <returns>The created <see cref="OpenApiResponses"/>.</returns>
-        public static OpenApiResponses CreateResponses(this ODataContext context, IEdmOperation operation)
+        public static OpenApiResponses CreateResponses(this ODataContext context, IEdmOperation operation, ODataPath path)
         {
             Utils.CheckArgumentNull(context, nameof(context));
             Utils.CheckArgumentNull(operation, nameof(operation));
+            Utils.CheckArgumentNull(path, nameof(path));
 
-            OpenApiResponses responses = new OpenApiResponses();
+            OpenApiResponses responses = new();
 
             if (operation.IsAction() && operation.ReturnType == null)
             {
                 responses.Add(Constants.StatusCode204, Constants.StatusCode204.GetResponse());
             }
             else
             {
-                OpenApiResponse response = new OpenApiResponse
+                OpenApiSchema schema;
+                if (operation.ReturnType.TypeKind() == EdmTypeKind.Collection)
+                {
+                    // Get the entity type of the previous segment
+                    IEdmEntityType entityType = path.Segments.Reverse().Skip(1)?.Take(1)?.FirstOrDefault()?.EntityType;
+                    schema = new OpenApiSchema
+                    {
+                        Title = entityType == null ? null : $"Collection of {entityType.Name}",
+                        Type = "object",
+                        Properties = new Dictionary<string, OpenApiSchema>
+                        {
+                            {
+                                "value", context.CreateEdmTypeSchema(operation.ReturnType)
+                            }
+                        }
+                    };
+                }
+                else
+                {
+                    // A property or operation response that is of a primitive type is represented as an object with a single name/value pair,
+                    // whose name is value and whose value is a primitive value.
+                    if (operation.ReturnType.IsPrimitive())
```

Comment:
nit: This can be simplified with an `else if` since we are also checking the `operation.ReturnType` in the `if` above.